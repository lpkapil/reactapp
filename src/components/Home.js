import React from 'react';
import PageHeaderCommon from '../components/Layout/PageHeaderCommon';
import { Skeleton, Switch, List, Avatar } from 'antd';
import { StarOutlined, LikeOutlined, MessageOutlined } from '@ant-design/icons';

const listData = [];
for (let i = 1; i < 2; i++) {
  listData.push({
    href: 'https://www.google.com',
    title: `Test Tile ${i}`,
    avatar: 'https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png',
    description:
      'Lorem Ipsum is simply dummy text.',
    content:
      'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrysstandard dummy text ever since the 1500s.',
  });
}

const IconText = ({ icon, text }) => (
  <span>
    {React.createElement(icon, { style: { marginRight: 8 } })}
    {text}
  </span>
);

class Home extends React.Component {

  constructor(props){
    super(props);
    this.state = {
      loading: true,
    };
  }
  

  onChange = checked => {
    this.setState({ loading: !checked });
  };

  render() {
    return (
      <div className="Home">
        <PageHeaderCommon title="Home" subTitle="Home page" />
        <p>Progressive content <Switch checked={!this.state.loading} onChange={this.onChange} /></p>
        

        <List
          itemLayout="vertical"
          size="large"
          dataSource={listData}
          renderItem={item => (
            <List.Item
              key={item.title}
              actions={
                !this.state.loading && [
                  <IconText icon={StarOutlined} text="156" key="list-vertical-star-o" />,
                  <IconText icon={LikeOutlined} text="156" key="list-vertical-like-o" />,
                  <IconText icon={MessageOutlined} text="2" key="list-vertical-message" />,
                ]
              }
              extra={
                !this.state.loading && (
                  <img
                    width={300}
                    height={200}
                    alt="logo"
                    src="https://gw.alipayobjects.com/zos/rmsportal/mqaQswcyDLcXyDKnZfES.png"
                  />
                )
              }
            >
              <Skeleton loading={this.state.loading} active avatar>
                <List.Item.Meta
                  avatar={<Avatar src={item.avatar} />}
                  title={<a href={item.href}>{item.title}</a>}
                  description={item.description}
                />
                {item.content}
              </Skeleton>
            </List.Item>
          )}
        />
      </div>
    )
  }
}

export default Home;
