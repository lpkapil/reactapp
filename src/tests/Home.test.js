import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer'; 
import Home from '../components/Home';

test('renders home page content', () => {
  // const { getByText } = render(<Home />);
  // const linkElement = getByText(/This is home page/i);
  // expect(linkElement).toBeInTheDocument();
});

describe('Homepage snapshot', () => {
  test('Homepage snapshot test', () => {
    const component = renderer.create(<Home />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

