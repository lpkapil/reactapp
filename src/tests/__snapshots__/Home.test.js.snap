// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage snapshot Homepage snapshot test 1`] = `
<div
  className="Home"
>
  <div>
    <div
      className="ant-row"
      style={
        Object {
          "marginBottom": 8,
          "marginLeft": -8,
          "marginRight": -8,
          "marginTop": -8,
        }
      }
    >
      <div
        className="ant-col ant-col-24"
        style={
          Object {
            "paddingBottom": 8,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 8,
          }
        }
      >
        <div
          className="ant-page-header site-page-header ant-page-header-ghost"
        >
          <div
            className="ant-page-header-heading"
          >
            <div
              className="ant-page-header-heading-left"
            >
              <span
                className="ant-page-header-heading-title"
                title="Home"
              >
                Home
              </span>
              <span
                className="ant-page-header-heading-sub-title"
                title="Home page"
              >
                Home page
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <p>
    This is home page
  </p>
</div>
`;
